lib/Stan_math_backend.ml

```diff
@@ -2,77 +2,59 @@
 
 open Core_kernel
 open Mir
-open Format
+open Fmt
 
-let comma ppf () = fprintf ppf ",@ "
-let newline ppf () = fprintf ppf "@;"
-let semi_new ppf () = fprintf ppf "@;"
-let emit_str ppf s = fprintf ppf "%s" s
 let zero = Lit (Int, "0")
 
-let emit_option ppf (emitter, opt, default) =
-  match opt with
-  | Some x -> fprintf ppf "%a" emitter x
-  | None -> emit_str ppf default
-
-let rec emit_unsizedtype ad ppf = function
-  | Ast.UInt | UReal -> emit_str ppf ad
-  | UArray t -> fprintf ppf "std::vector<%a>" (emit_unsizedtype ad) t
-  | UMatrix -> fprintf ppf "Eigen::Matrix<%s, -1, -1>" ad
-  | URowVector -> fprintf ppf "Eigen::Matrix<%s, 1, -1>" ad
-  | UVector -> fprintf ppf "Eigen::Matrix<%s, -1, 1>" ad
+let rec pp_unsizedtype ad ppf = function
```

Comment:
Not sure how I feel about the naming `pp_`. To me, that sounds like pretty printing. How about `gen_` or `cpp_`?

Reply:
I'm following the convention for Fmt: https://erratique.ch/software/fmt/doc/Fmt.html#nameconv. And we are pretty printing C++.

Reply:
OK!